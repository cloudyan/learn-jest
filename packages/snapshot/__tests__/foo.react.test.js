import React from 'react';
import { create, act } from 'react-test-renderer';
import Foo from '../Foo.react';

// https://zh-hans.reactjs.org/docs/test-renderer.html

function InFoo(props) {
  return <div foo={props.foo} className="in-foo">InFoo</div>;
}

function Bar() {
  return (
    <div className="in-bar">
      <InFoo foo="bar" />
    </div>
  );
}

// jest.dontMock('..');

describe('<Bar />', () => {
  it('renders correctly', () => {
    const tree = create(<div className="title">xxx</div>);
    expect(tree).toMatchSnapshot();
  });

  it('test1', () => {
    const tree = create(<Bar />);
    expect(tree).toMatchSnapshot();

    const rootInstance = tree.root;
    expect(rootInstance.findByType(InFoo).props.foo).toBe('bar');
    expect(rootInstance.findByProps({className: "in-foo"}).children).toEqual(['InFoo']);
  });
});

describe('<Foo />', () => {
  it('should render without throwing an error', () => {
    const tree = create(<Foo title="First Demo" />);
    expect(tree).toMatchSnapshot();
  });

  it('should render without throwing an error 1', () => {
    const tree = create(
      <Foo title="First Demo">
        <p className="p">test</p>
      </Foo>,
    );
    expect(tree).toMatchSnapshot();
  });
});
