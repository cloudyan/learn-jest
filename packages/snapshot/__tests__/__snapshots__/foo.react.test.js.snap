// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bar /> renders correctly 1`] = `
<div
  className="title"
>
  xxx
</div>
`;

exports[`<Bar /> test1 1`] = `
<div
  className="in-bar"
>
  <div
    className="in-foo"
    foo="bar"
  >
    InFoo
  </div>
</div>
`;

exports[`<Foo /> should render without throwing an error 1`] = `
<h1
  className="title"
>
  First Demo
</h1>
`;

exports[`<Foo /> should render without throwing an error 1 1`] = `
<h1
  className="title"
>
  First Demo
  <p
    className="p"
  >
    test
  </p>
</h1>
`;
